common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockTransferor.java

```diff
@@ -192,8 +197,12 @@ private synchronized void initiateRetry() {
   private synchronized boolean shouldRetry(Throwable e) {
     boolean isIOException = e instanceof IOException
       || e.getCause() instanceof IOException;
+    boolean isSaslTimeout = enableSaslRetries &&
+        (e instanceof TransportClient.SaslTimeoutException ||
+            (e.getCause() != null && e.getCause() instanceof TransportClient.SaslTimeoutException));
```

Comment:
With the proposed changes, this will simply become:
```suggestion
        (e instanceof TransportClient.SaslTimeoutException);
```